client/components/ReviewsWidget/ReviewsWidget.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import axios from 'axios';
+import ReviewTab from '../ReviewTab/ReviewTab';
+import Carousel from '../Carousel/Carousel';
+import ReviewPager from '../ReviewPager/ReviewPager';
+import MainHeader from '../MainHeader/MainHeader';
+import ReviewsWidgetContainer from './ReviewsWidget.styles';
+
+const serverUrl = 'http://localhost:8888';
+
+export default class ReviewsWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      storeReviewCount: 0,
+      rating: 0,
+      productReviewCount: 0,
+      reviewList: [],
+      reviewPictures: [],
+      pageNumber: 1,
+      sortBy: 'rating',
+      isDropdownOpen: false,
+      activeTab: 'productReviews',
+    };
+    this.handlePageClick = this.handlePageClick.bind(this);
+    this.handleSortByClick = this.handleSortByClick.bind(this);
+    this.handleDropdownClick = this.handleDropdownClick.bind(this);
+    this.handleMenuClick = this.handleMenuClick.bind(this);
+  }
+
+  componentDidMount() {
+    const id = ReviewsWidget.getId();
+    Promise.all([
+      axios.get(`${serverUrl}/review-summary/${id}`),
+      axios.get(`${serverUrl}/review-list/${id}`),
+      axios.get(`${serverUrl}/reviews-pictures/${id}`),
+    ])
+      .then(([reviewSummary, reviewList, reviewPictures]) => {
+        this.setState({
+          ...reviewSummary.data,
+          reviewList: reviewList.data,
+          reviewPictures: reviewPictures.data,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+
+  static getId() {
+    const url = window.location.pathname;
+    const id = url.substring(url.lastIndexOf('/') + 1) || 1;
+    return id;
+  }
+
+  getReviews(id, pageNumber, sortBy, store) {
+    axios.get(`${serverUrl}/review-list/${id}`, {
+      params: {
+        pageNumber,
+        sortBy,
+        store,
+      },
+    })
+      .then((response) => {
+        this.setState({
+          reviewList: response.data,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+
+  handleSortByClick(e) {
+    e.preventDefault();
```

Comment:
Out of curiosity what default action are you trying to prevent?

Reply:
😇  eagle eye. thanks!!